### PR TITLE
Extract adjuncts from Vertex class

### PR DESCRIPTION
**Pull request by rmilfont**
Now Vertex class are more simple. The adjunct's list and other useful
methods (involving adjuncts) can be find from Graph class.
Commit:Mod
TaskNumber:#57

### CODE REVIEW

File: src/util/LinkWeightTracker.java [hunk 1]
```diff
@@ -64,4 +66,12 @@ public double weight(Path p) {
     public Collection<Double> edgesWeight() {
         return edgesWeights.values();
     }
+
+    private Collection<Edge> getEdgesOf(Path p){
```
Comment (jmferreiratech) [inline src/util/LinkWeightTracker.java]: Rename it to edgesOf

File: src/rbr/RBR.java [hunk 2]
```diff
@@ -285,7 +285,8 @@ private boolean reaches(Vertex src, Vertex dest, Character ipColor) {
 		if (opColor == null) {
 			return false;
 		}
-		return reaches(src.adjunct(opColor).destination(), dest, EdgeColor.getInvColor(src.adjunct(opColor).color()));
+		Edge newSrc = graph.getAdjunct(src, opColor);
```
Comment (jmferreiratech) [inline src/rbr/RBR.java]: "newSrc" sounds weird to me.

File: src/util/Graph.java [hunk 3]
```diff
@@ -40,12 +42,37 @@ private void reachAdjuncts(Vertex vertex, Set<Vertex> reachable) {
 		return this.edges;
 	}
 
+    public ArrayList<Edge> getAdjuncts(Vertex v) {
```
Comment (jmferreiratech) [inline src/util/Graph.java]: rename it to adjunctsOf(Vertex src) or only adjuncts(Vertex src)

File: src/util/Graph.java [hunk 4]
```diff
@@ -40,12 +42,37 @@ private void reachAdjuncts(Vertex vertex, Set<Vertex> reachable) {
...
+        return this.adjuncts.get(v);
+    }
+
+    public Edge getAdjunct(Vertex src, Vertex dst){
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Rename it to adjunct(Vertex from, Vertex to)

File: src/util/Graph.java [hunk 5]
```diff
@@ -40,12 +42,37 @@ private void reachAdjuncts(Vertex vertex, Set<Vertex> reachable) {
...
+        for(Edge edge : getAdjuncts(src)){
+            if (edge.destination().name().equals(dst.name())) {
+                return edge;
+            }
+        }
+        System.out.println("ERROR : There isn't adjunct between " + src.name() + "and" + dst.name());
+        return null;
+    }
+
+    public Edge getAdjunct(Vertex v, Character color){
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Rename it to adjunctOf()

File: src/util/Graph.java [hunk 6]
```diff
@@ -5,12 +5,14 @@
 public class Graph {
 	private ArrayList<Vertex> vertices;
 	private ArrayList<Edge> edges;
+    private Map<Vertex, ArrayList<Edge>> adjuncts;
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Use a more abstract type instead of ArrayList

File: src/util/Graph.java [hunk 7]
```diff
@@ -59,30 +86,41 @@ public Vertex vertex(String name) {
...
 		Vertex v = new Vertex(nome);
 		vertices.add(v);
+        adjuncts.put(v, new ArrayList<>());
 	}
 
-	void addEdge(Vertex origem, Vertex destino, Character cor) {
-		Edge e = new Edge(origem, destino, cor);
-		origem.addAdjunct(e);
+	void addEdge(Vertex src, Vertex dst, Character cor) {
+		Edge e = new Edge(src, dst, cor);
```
Comment (jmferreiratech) [inline src/util/Graph.java]: Here you should call only addEdge(new Edge(src, dst, cor));

File: src/util/Graph.java [hunk 8]
```diff
@@ -59,30 +86,41 @@ public Vertex vertex(String name) {
...
 		edges.add(toAdd);
 	}
 
 	void removeEdge(Edge toRemove) {
-		toRemove.source().adjuncts().remove(toRemove);
+        adjuncts.get(toRemove.source()).remove(toRemove);
 		edges.remove(toRemove);
 	}
 
+    void findAdjuncts(){
```
Comment (jmferreiratech) [inline src/util/Graph.java]: I cannot see why you need this method since addEdge() do this job already.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

File: src/util/LinkWeightTracker.java (hunk 1 above)
Comment (rmilfont) [reply src/util/LinkWeightTracker.java]: Done.

File: src/util/Graph.java (hunk 6 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 5 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 4 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 8 above)
Comment (rmilfont) [reply src/util/Graph.java]: With the above suggestion, I will remove the method.

File: src/util/Graph.java (hunk 7 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/util/Graph.java (hunk 3 above)
Comment (rmilfont) [reply src/util/Graph.java]: Done.

File: src/rbr/RBR.java (hunk 2 above)
Comment (rmilfont) [reply src/rbr/RBR.java]: Done.

**Comment by rmilfont** [comment]
All done.

**Review by jmferreiratech — APPROVED** [review]

**Pull request merged**